src/equations/jin_xin_compressible_euler_2d.jl

```diff
@@ -6,190 +6,192 @@
+        dissipation = SVector((sqrt_velocities .* (u_rr_base - u_ll_base))...,
+                              (sqrt_velocities .* (f_rr_base + f_ll_base))...,
+                              zero(u_ll_base)...)
+    else # orientation == 2
+        sqrt_velocities = equations.sqrt_velocities[2]
+        f_ll_base = get_block_components(u_ll, 3, equations)
+        f_rr_base = get_block_components(u_rr, 3, equations)
+        dissipation = SVector((sqrt_velocities .* (u_rr_base - u_ll_base))...,
+                              zero(u_ll_base)...,
+                              (sqrt_velocities .* (f_rr_base + f_ll_base))...)
     end
-    return 0.5 * (flux(u_ll,orientation,equations) + flux(u_rr,orientation,equations) - dissipation)
+
+    return 0.5f0 * (flux(u_ll, orientation, equations) +
+                    flux(u_rr, orientation, equations) - dissipation)
 end
 
 
-"""                                                                                                 
-    collision_bgk(u, dt, equations::LatticeBoltzmannEquations2D)                                    
-                                                                                                    
-Collision operator for the Bhatnagar, Gross, and Krook (BGK) model.                                 
-"""                                                                                                 
-@inline function collision_bgk(u, dt, equations::JinXinCompressibleEulerEquations2D)                       
-    # relaxation parameter
-    eps = equations.eps_relaxation
-
-    # compute compressible Euler fluxes
-    u_cons = SVector(u[1], u[2], u[3], u[4])
-    eq_relax = equations.equations_relaxation
-    vu = flux(u_cons,1,eq_relax)
-    wu = flux(u_cons,2,eq_relax)
-
-    dt_ = dt * 1.0
-    factor =1.0/ (eps + dt_)
-
-    # compute relaxation terms
-    du1 = 0.0
-    du2 = 0.0
-    du3 = 0.0
-    du4 = 0.0
-    du5 = -u[5] + factor * (eps * u[5] + dt_ * vu[1])
-    du6 = -u[6] + factor * (eps * u[6] + dt_ * vu[2])
-    du7 = -u[7] + factor * (eps * u[7] + dt_ * vu[3])
-    du8 = -u[8] + factor * (eps * u[8] + dt_ * vu[4])
-    du9 = -u[9] + factor * (eps * u[9] + dt_ * wu[1])
-    du10= -u[10]+ factor * (eps * u[10]+ dt_ * wu[2])
-    du11= -u[11]+ factor * (eps * u[11]+ dt_ * wu[3])
-    du12= -u[12]+ factor * (eps * u[12]+ dt_ * wu[4])
-                                                                                                    
-    return SVector(du1, du2, du3, du4, du5, du6, du7, du8, du9, du10, du11, du12)                                                              
-end  
-
-
-
-@inline function max_abs_speeds(u,equations::JinXinCompressibleEulerEquations2D)
-    sa = equations.sqrt_a
-    sb = equations.sqrt_b
-
-    return max(sa[1],sa[2],sa[3],sa[4]), max(sb[1],sb[2],sb[3],sb[4])
+@inline function max_abs_speeds(u, equations::JinXinEquations{2})
+    return ntuple(Val(ndims(equations))) do n
+        maximum(equations.sqrt_velocities_inv[n])
```

Comment:
```suggestion
        maximum(equations.sqrt_velocities[n])
```
Again, was this by accident or do you really think it should be the inverse of the sqrt of the velocities?

Reply:
No, should be without `inv` - looks like I hacked this stuff together too quickly